s2and/model.py

```diff
@@ -962,40 +978,51 @@ def __init__(self, linkage: str = "average", eps: float = 0.5, preserve_input: b
         self.linkage = linkage
         self.eps = eps
         self.preserve_input = preserve_input
+        self.input_as_observation_matrix = input_as_observation_matrix
         self.labels_ = None
 
     def fit(self, X: np.array) -> np.array:
         """
         Fit the estimator on input data. The results are stored in self.labels_.
-
         Parameters
         ----------
         X: np.array
             The input may be either a 1-D condensed distance matrix
             or a 2-D array of observation vectors. If X is a 1-D condensed distance
             matrix, then it must be (N choose 2) sized vector, where N is the number
-            of original observations paired in the distance matrix.
-
+            of original observations paired in the distance matrix. If X is 2-D
+            then the flag `input_as_observation_matrix` must be set to True in init.
         Returns
         -------
-            self
+        self
         """
+        X = np.asarray(X)
+        if len(X.shape) == 1 and self.input_as_observation_matrix:
+            raise Exception(
+                "Input to fit is one-dimensional, but input_as_observation_matrix flag is set to True. "
+                "If you intended to pass in a distance matrix, it must be flattened (1-D)"
```

Comment:
these "If you..."s are reversed for which error they should be printed for

Reply:
Oops...